tests/thread_cooperation/tests/01-run.py

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+# Copyright (C) 2017 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import sys
+
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
+
+def testfunc(child):
+    # 1st 6LoWPAN fragment
+    child.expect(r"\[START\].*\n")
```

Comment:
The python expect could check more the test application output.

Reply:
why? (IMHO) It just makes the script longer and complicated with no additional use, if no `[SUCCESS]` shows up at the end the test failed, parsing any other output doesn't make sense here. It makes sense for some of our timer tests to check if certain jitter/bounds hold, but that doesn't hold here.

Reply:
with #7845 in, this will help to find a problem if there's any. But I also admit that it somehow duplicate the displayed information between C and Python.

Reply:
This line could be changed into:
```python
child.expect(r"\[START\]*.")
```
I don't think the `\n` is needed and the end of line regexp should be `*.` instead of `.*`.

Maybe it could be interesting to check that the test application returns the right factorial result ? (one can compare with the result of:
```python
math.factorial(12)
```
in the pexpect script

Reply:
> check that the test application returns the right factorial 

that's already done in the C code of main, could move to test script but then you don't get the result when just running `make term`

Reply:
> and the regexp should be *. instead of .*

? but then its no regex anymore, at least not as I know them. IIRC `*` means 0 or more of whatever expression is before, and `.` means _any valid unicode_, hence `.*` means anything or nothing. Or `\d+` means "any number with at least one digit`. But maybe misunderstood regex?

Reply:
I think your understanding is correct, I'm thinking the same, but inverting them makes the test pass for me.

Reply:
> that's already done in the C code of main, could move to test script but then you don't get the result when just running make term

agreed, so let's keep it as it is.